v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -476,53 +508,73 @@ func (r *AzureDeploymentReconciler) handlePollerFailed(ctx context.Context, err
 	return ctrl.Result{Requeue: !isFatal}, nil
 }
 
-func (r *AzureDeploymentReconciler) handlePollerSuccess(ctx context.Context) (ctrl.Result, error) {
+func (r *AzureDeploymentReconciler) makeSuccessResult() ctrl.Result {
+	result := ctrl.Result{}
+	// This has a RequeueAfter because we want to force a re-sync at some point in the future in order to catch
+	// potential drift from the state in Azure. Note that we cannot use mgr.Options.SyncPeriod for this because we filter
+	// our events by predicate.GenerationChangedPredicate and the generation will not have changed.
+	if r.config.SyncPeriod != nil {
+		result.RequeueAfter = randextensions.Jitter(r.rand, *r.config.SyncPeriod, 0.1)
+	}
+	return result
+}
+
+func (r *AzureDeploymentReconciler) handleSkipReconcile(ctx context.Context) (ctrl.Result, error) {
+	reconcilePolicy := r.GetReconcilePolicy()
 	r.log.V(Status).Info(
-		"Resource successfully created",
+		"Skipping creation of resource due to policy",
+		ReconcilePolicyAnnotation, reconcilePolicy,
 		"resourceID", genruntime.GetResourceIDOrDefault(r.obj))
 
-	ready := r.PositiveConditions.Ready.Succeeded(r.obj.GetGeneration())
-	resourceID, hasResourceID := genruntime.GetResourceID(r.obj)
-	if !hasResourceID {
-		return ctrl.Result{}, errors.Errorf("poller succeeded but resource has no resource id")
+	err := r.updateStatus(ctx)
+	if err != nil {
+		if genericarmclient.IsNotFoundError(err) {
+			conditions.SetCondition(
+				r.obj,
+				r.PositiveConditions.Ready.AzureResourceNotFound(r.obj.GetGeneration(), err.Error()))
+			r.ClearPollerResumeToken()
+			err = r.CommitUpdate(ctx)
+			if err != nil {
+				// NotFound is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+				// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		}
+		return ctrl.Result{}, err
 	}
 
-	// TODO: I think if we're really we can avoid this as the HTTP payload in poller should have all we need already?
-	// TODO: for now doing this though because my quick read of the Go SDK is that it does a final GET on the URL too...
-	// TODO: Maybe some Azure services or types of async operations actually don't return the latest resource?
-	status, _, err := r.getStatus(ctx, resourceID)
+	r.ClearPollerResumeToken()
+	conditions.SetCondition(r.obj, r.PositiveConditions.Ready.Succeeded(r.obj.GetGeneration()))
+	err = r.CommitUpdate(ctx)
 	if err != nil {
```

Comment:
Here we can use something like `if err = r.CommitUpdate(ctx) ; err != nil {`

Reply:
Done